src/QMCDrivers/WalkerLogManager.cpp

```diff
@@ -0,0 +1,360 @@
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2024 QMCPACK developers.
+//
+// File developed by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#include "WalkerLogManager.h"
+#include "WalkerLogInput.h"
+#include "Concurrency/OpenMP.h"
+
+
+namespace qmcplusplus
+{
+
+
+WalkerLogManager::WalkerLogManager(WalkerLogInput& inp, bool allow_logs, std::string series_root, Communicate* comm)
+{
+  state.reset();
+  communicator            = comm;
+  file_root               = series_root;
+  bool driver_allows_logs = allow_logs; // driver allows logs or not
+
+  bool logs_requested = inp.present; // xml input present or not
+  // determine whether walker logs will be active
+  state.logs_active = logs_requested && driver_allows_logs;
+
+  if (state.logs_active)
+  {
+    if (omp_get_thread_num() == 0)
+    {
+      app_log() << "\n  WalkerLogManager::put() " << std::endl;
+      app_log() << "    logs requested      : " << logs_requested << std::endl;
+      app_log() << "    driver allows logs  : " << driver_allows_logs << std::endl;
+      app_log() << "    logs active         : " << state.logs_active << std::endl;
+      app_log() << std::endl;
+    }
+    // retrieve input data
+    state.step_period   = inp.get<int>("step_period");
+    state.verbose       = inp.get<bool>("verbose");
+    bool quantiles      = inp.get<bool>("quantiles");
+    write_particle_data = inp.get<bool>("particle");
+    write_min_data      = inp.get<bool>("min") && quantiles;
+    write_max_data      = inp.get<bool>("max") && quantiles;
+    write_med_data      = inp.get<bool>("median") && quantiles;
+  }
+
+  // label min energy walker buffers for HDF file write
+  wmin_property_int_buffer.label  = "wmin_property_int";
+  wmin_property_real_buffer.label = "wmin_property_real";
+  wmin_particle_real_buffer.label = "wmin_particle_real";
+
+  // label max energy walker buffers for HDF file write
+  wmax_property_int_buffer.label  = "wmax_property_int";
+  wmax_property_real_buffer.label = "wmax_property_real";
+  wmax_particle_real_buffer.label = "wmax_particle_real";
+
+  // label median energy walker buffers for HDF file write
+  wmed_property_int_buffer.label  = "wmed_property_int";
+  wmed_property_real_buffer.label = "wmed_property_real";
+  wmed_particle_real_buffer.label = "wmed_particle_real";
+
+  // walker quantity information ("data_layout") has not be put in the HDF file yet
+  registered_hdf = false;
+}
+
+
+WalkerLogCollector* WalkerLogManager::makeCollector()
+{
+  if (state.verbose)
+    app_log() << "WalkerLogManager::makeCollector " << std::endl;
+  WalkerLogCollector* tc = new WalkerLogCollector();
+  tc->state              = state;
```

Comment:
Makes no difference, but I'll spend extra time on it anyway.

Reply:
Will make this change.

Reply:
Done

Reply:
I cleaned up the constructor of `WalkerLogCollector` in https://github.com/QMCPACK/qmcpack/pull/5039. Only the one with state got kept. All the collectors should be created via the manager since state is no longer a copy but a reference to the manager state.